### ISSUE TITLE
Allow execution of zero parameter functions

### ISSUE THREAD

**Issue by Jimexist** [apache/arrow-datafusion#305]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

We shall allow the execution of zero parameter functions.

**Describe the solution you'd like**

One solution is to conventionally pass a null array with the correct expected batch size to `ScalarFunctionImplementation` so that functions with zero param can take that as input.

where the first param `usize` is the batch size

**Describe alternatives you've considered**

**Additional context**

**Comment by jorgecarleitao** [comment]
Some alternatives:

* create a node for these type of operations.

* have the physical planner pass a `NullArray` to functions of zero arguments; `NullArray` has a length (only has a length).

**Comment by Jimexist** [comment]
> It is not obvious to me that changing the signature of ScalarFunctionImplementation is the right approach here.
Agree on this.

Let me rephrase
